enterprise/frontend/src/metabase-enterprise/license/use-upsell-flow.ts

```diff
@@ -98,9 +98,12 @@ function createListener({
   storeOrigin,
 }: {
   updateToken: (token: string) => Promise<void>;
-  storeOrigin: string;
+  storeOrigin: string | undefined;
 }) {
   return (event: MessageEvent<LicenseTokenMessage>) => {
+    if (!storeOrigin) {
+      return;
+    }
```

Comment:
instead of handling the empty case, can we define a fallback value in the selector?

Reply:
We could, and that would simplify the code a bit, but we would be lying a bit. Like: here's the `storeUrl`, and then in 100ms later, sorry, actually it's a different `storeUrl`. The consumer doesn't know when the value has stabilized. And what would should be this URL? Production store URL for production and staging envs, or the other way around, or some other, unrelated URL. And the truth is that, since we agreed that source of truth for that URL is on the BE side, initially we don't know what's the URL (`undefined`), and after brief milliseconds we know for sure. Maybe I'm getting too theoretical 😀

And on practical level that could also cause some issues in tests, like unexpected changes for assertions that check for the URL. 🤔